vllm/model_executor/models/internvl.py

```diff
@@ -175,13 +175,15 @@ def get_internvl_num_patches(image_size: int, patch_size: int,
         (downsample_ratio**2))
 
 
-def get_max_internvl_image_tokens(ctx: InputContext):
+def get_max_internvl_image_tokens(ctx: InputContext,
+                                  *,
+                                  max_dynamic_patch: Optional[int] = None):
     hf_config = ctx.get_hf_config()
     vision_config = hf_config.vision_config
 
     use_thumbnail = hf_config.use_thumbnail
-    max_dynamic_patch = hf_config.max_dynamic_patch
-    if use_thumbnail:
+    max_dynamic_patch = max_dynamic_patch or hf_config.max_dynamic_patch
```

Comment:
Let's use an explicit `None` check to avoid silently using HF's config when `max_dynamic_patch == 0`.